addons/mail/static/src/core/common/persona_model.js

```diff
@@ -56,26 +56,21 @@ export class Persona extends Record {
     storeAsTrackedImStatus = fields.One("Store", {
         /** @this {import("models").Persona} */
         compute() {
-            if (
-                this.type === "guest" ||
-                (this.type === "partner" && this.im_status !== "im_partner" && !this.is_public)
```

Comment:
How's the check on token replacing the check on im_partner or public?
Won't we send a token for those regardless?

Reply:
I think we should only send the token when it makes sense? This way it's dead simple: you have the token, you subscribe.

Reply:
The token is just there to grant access, it's not there to make business decisions on whether a given presence should be subscribed or not.